user-guide/Advanced_Modules/DOM/DOM_best_practices.md

```diff
@@ -0,0 +1,200 @@
+
+In short, the following is important:
+
+- **Group DomDefinitions:** Consolidate related `DomDefinitions` within the same module to optimize database performance and minimize resource usage.
+
+- **Disable Instance History:** Turn off unnecessary instance history tracking to conserve resources, especially beneficial for DOM managers with frequent create or update actions.
+
+- **Compact DomInstances:** Prioritize essential data, avoid duplication, and limit unnecessary fields to keep `DomInstances` small and efficient.
+
+- **Optimize CRUD Calls:** Use bulk operations for create, update, or delete actions, and avoid multiple updates to the same `DomInstance` in the same script to reduce overhead.
+
+- **Efficient Scripting:** Design scripts for fast and efficient execution, particularly focusing on CRUD scripts, to minimize execution time and potential script queues.
+
+## Module and settings
+
+### Group DomDefinitions that belong to the same solution within the same DOM module
+
+It is recommended to limit the number of DOM modules on a DataMiner system. Each DOM module has its separate indexes in the Elasticsearch or OpenSearch database. These require a certain amount of heap memory and add to the number of active 'shards' in the database. It is thus recommended to define all `DomDefinitions` and store all `DomInstances` within the same DOM module instead of creating a DOM module for each separate DOM model. However, if a certain model is completely stand-alone and is reused by other solutions, it can still be advisable to keep them separate. It is recommended to keep the number of DOM modules on one system well below 50. This number is, however, dependent on the number of DB nodes and their resource allocation. If STaaS is used, the performance aspect is less relevant due to the different way the DOM data is stored.
+
+**Not recommended:**
+
+- Module: *people_app_offices*
+  - Definition: Offices
+- Module: *people_app_grouping*
+  - Definition: Teams
+  - Definition: Departments
+- Module: *people_app_people*
+  - Definition: People
+
+**Recommended:**
+
+- Module: *people_app*
+  - Definition: Offices
+  - Definition: Teams
+  - Definition: Departments
+  - Definition: People
+
+### Disable DomInstance history when not required
+
+By default, any change made to a `DomInstance` is tracked in [history records](xref:DOM_history). These records are asynchronously saved by default, which should limit the performance impact when saving or updating a `DomInstance`. However, saving these history records puts additional load on both DataMiner and the database. If history is not required, it is recommended to [disable](xref:DOM_history#disabling-the-history-or-changing-the-storage-behavior) it to save CPU cycles, memory, and storage. For DOM managers experiencing frequent create or update actions (> 1 per second on average), the performance benefit of disabling this history could be considerable.
```

Comment:
There is only one afaik, which is the stepper component. But this is already made to handle this case. When there is no history enabled, it will still work, but it won't show the exact state history but only the expected previous states according to the state diagram. Looks like it is already documented (second note): https://docs.dataminer.services/user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/Visualizations/Available_visualizations/General/Stepper.html.